tests/test_task.py

```diff
@@ -0,0 +1,174 @@
+        counts["a"] / len(modality_seq),
+        task.session["a"] - task.catch_prob * task.session["a"],
+        atol=0.1,
+    )
+
+
+def test__build_trials_seq_shuffling():
+    task_shuffled = Task(NAME, shuffle_trials=True)
+    task_not_shuffled = Task(NAME, shuffle_trials=False)
+
+    task_shuffled._ntrials = NTRIALS  # noqa: SLF001
+    task_not_shuffled._ntrials = NTRIALS  # noqa: SLF001
+
+    task_shuffled._rng = np.random.default_rng(NTRIALS)  # noqa: SLF001
+    task_not_shuffled._rng = np.random.default_rng(NTRIALS)  # noqa: SLF001
+
+    sequence_shuffled = task_shuffled._build_trials_seq()  # noqa: SLF001
+    sequence_not_shuffled = task_not_shuffled._build_trials_seq()  # noqa: SLF001
+
+    # Verify that the generated sequences are shuffled or not shuffled accordingly
+    assert sequence_shuffled.shape == sequence_not_shuffled.shape
+    assert not np.array_equal(sequence_shuffled, sequence_not_shuffled)
+
+
+def test__build_trials_seq_maximum_sequential_trials():
+    # Create a Task instance with shuffling enabled and a maximum sequential trial constraint
+    task = Task(name=NAME, max_sequential=4)
+    task._ntrials = NTRIALS  # noqa: SLF001
+    task._rng = np.random.default_rng(NTRIALS)  # noqa: SLF001
+    modality_seq = task._build_trials_seq()  # noqa: SLF001
+    # Ensure that no more than the specified maximum number of consecutive trials of the same modality occur
+    for modality in task.modalities:
+        for i in range(len(modality_seq) - task.max_sequential):
+            assert np.sum(modality_seq[i : i + task.max_sequential] == modality) <= task.max_sequential
+
+
+def test_generate_trials(task: Task):
+    trials = task.generate_trials()
+    assert trials["inputs"].shape == (task._ntrials, len(task.time), task.n_inputs)  # noqa: SLF001
+    assert trials["outputs"].shape == (task._ntrials, len(task.time), task.n_outputs)  # noqa: SLF001
+
+
+def test_plot_trials(task: Task):
+    # Generate trial data
+    ntrials = 3
+    task.generate_trials(ntrials=ntrials)
+    # Call plot_trials
+    n_plots = 2
+    fig = task.plot_trials(n_plots=n_plots)
+    # Assert basic properties of the plot
+    assert isinstance(fig, go.Figure)
+    assert len(fig["data"]) / (task.n_inputs + task.n_outputs) == n_plots  # Number of plots should match n_plots
+    assert fig.layout.title.text == "Trials"  # Check title
+    # Test with n_plots > ntrials
+    n_plots = 5
+    with pytest.warns(UserWarning) as w:
+        fig = task.plot_trials(n_plots=n_plots)
+    # Check if the warning was raised
+    assert len(w) == 1
+    warning = w[0]
+    assert (
+        str(warning.message)
+        == f"Number of plots requested ({n_plots}) exceeds number of trials ({ntrials}). Will plot all trials."
+    )
+    assert warning.category == UserWarning
+    assert isinstance(fig, go.Figure)
+    assert len(fig["data"]) / (task.n_inputs + task.n_outputs) == ntrials
+    assert fig.layout.title.text == "Trials"  # Check title
```

Comment:
Do we need to re-check the ones I removed in the `n_plots > ntrials` case?
```suggestion
    assert len(fig["data"]) / (task.n_inputs + task.n_outputs) == ntrials
```